models/articles-model.js

```diff
@@ -1,33 +1,93 @@
-const db = require('../db/connection');
+const db = require("../db/connection");
 
 exports.fetchArticleByID = (article_id) => {
-    return db
-    .query(`
+  return db
+    .query(
+      `
     SELECT articles.*, COUNT(comments.author) AS comment_count
     FROM articles
     LEFT JOIN comments
     ON articles.article_id = comments.article_id
     WHERE articles.article_id = $1
-    GROUP BY articles.article_id  
-    `, [article_id])
-    .then(({rows: articleData}) => {
-        if(articleData.length === 0) {
-            return Promise.reject({status: 404, msg: 'Article not found'});
-        }
-        else {
-            return articleData;
-        }
-    })
-}
+    GROUP BY articles.article_id 
+    `,
+      [article_id]
+    )
+    .then(({ rows: articleData }) => {
+      if (articleData.length === 0) {
+        return Promise.reject({ status: 404, msg: "Article not found" });
+      } else {
+        return articleData;
+      }
+    });
+};
 
-exports.updateArticleByID = (article_id, inc_votes) => {
-    if(typeof inc_votes !== 'number') {
-        return Promise.reject({status: 400, msg: 'Invalid Data Type'});
-    }
-    return db
-    .query(`UPDATE articles SET votes = (votes + $1) WHERE article_id = $2 RETURNING *;`, [inc_votes, article_id])
-    .then(({rows: [updatedArticleData]}) => {
-        return updatedArticleData
-    })
-}
+exports.fetchAllArticles = (topic) => {
+  const acceptableQueries = [];
 
+  let filteredQueryString = `SELECT articles.*, COUNT(comments.author) AS comment_count
+    FROM articles
+    LEFT JOIN comments
+    ON articles.article_id = comments.article_id
+    WHERE articles.topic = $1
```

Comment:
as this line is the only difference between the two, I suggest you have a look in the notes [here](https://notes.northcoders.com/courses/js-back-end/complex-queries) under 'optional WHERE clauses' to find a means of DRYing up your code